### PR TITLE
Separate parsing and chunking in `read_doc`

### PR DESCRIPTION
**Pull request by mskarlin**
This allows users to implement their own chunking if they choose, while retaining the same parsing as in standard PQA implementations. 

PR optionally allows users to get metadata about the parsing or chunking, for persistence / experiment tracking.

### CODE REVIEW

File: paperqa/readers.py [hunk 1]
```diff
@@ -4,56 +4,70 @@
...
     pages: list[str] = []
     texts: list[Text] = []
-    for i, page in enumerate(pdfReader.pages):
-        split += page.extract_text()
-        pages.append(str(i + 1))
+    split: str = ""
+
+    if not isinstance(parsed_text.content, dict):
+        raise NotImplementedError(
+            "pdf chunking only implemented per-page, ParsedText.content must be a `dict`."
```
Comment (jamesbraza) [inline paperqa/readers.py:65]: Can you use an f-string to include the actual type of `parsed_text.content`?

```python
f"ParsedText.content must be a `dict`, not {type(parsed_text.content)}."
```

File: paperqa/readers.py [hunk 2]
```diff
@@ -152,25 +206,80 @@ def parse_code_txt(path: Path, doc: Doc, chunk_chars: int, overlap: int) -> list
...
     path: Path,
     doc: Doc,
     chunk_chars: int = 3000,
     overlap: int = 100,
     force_pypdf: bool = False,
-) -> list[Text]:
-    """Parse a document into chunks."""
+    parsed_text_only: bool = False,
+    include_metadata: bool = False,
+) -> list[Text] | ParsedText | tuple[list[Text], ParsedMetadata]:
```
Comment (jamesbraza) [inline paperqa/readers.py:217]: (per discussion in-person)

This complex return type arises from multiple `bool` configuration arguments, which will lead to installers of `paper-qa` (and within `paper-qa` itself) having to add `type: ignore` comments. An immediate (yet verbose) solution is using `typing.overload` to better express this signature.

However, in general, this complex signature is a code smell for a decomposition needing to take place. Maybe in the future we can decompose this better to make the code more fluent without requiring big signatures

**Review by jamesbraza — APPROVED** [review]

File: paperqa/readers.py (hunk 2 above)
Comment (mskarlin) [reply paperqa/readers.py:265]: Added! Was able to remove the type ignores from `docs.py` after implementing.

Comment (jamesbraza) [reply paperqa/readers.py:265]: https://youtu.be/E5KfzyO4feo?t=21

@mskarlin is now Tron at 0:21 after making this change

**Pull request merged**